krkn/scenario_plugins/node_actions/abstract_node_scenarios.py

```diff
@@ -143,4 +144,90 @@ def node_service_status(self, node, service, ssh_private_key, timeout):
\ No newline at end of file
+        pass
+    
+    def node_disk_failure_scenario(self, instance_kill_count, node, timeout, disk_path="/dev/sdb"):
+        """
+        Simulate full disk failure on a node by offlining the disk using sysfs.
+        Args:
+            instance_kill_count: Number of times to run the scenario.
+            node: Node name to target.
+            timeout: Timeout in seconds for node status checks.
+            disk_path: Path to the disk (e.g., /dev/sdb).
+        """
+        for _ in range(instance_kill_count):
+            affected_node = AffectedNode(node)
+            try:
+                logging.info(f"Starting full disk failure scenario on node {node} for disk {disk_path}")
+                
+                # Prevent accidental root disk corruption
+                if disk_path in ["/dev/sda", "/dev/vda", "/dev/xvda"]:
+                    logging.error(f"Cannot simulate failure on root disk {disk_path}")
+                    raise ValueError(f"Cannot simulate failure on root disk {disk_path}")
+            
+                # Validate disk existence
+                disk_name = disk_path.split('/')[-1]
+                result = runcommand.run(
+                    f"oc debug node/{node} -- chroot /host ls /sys/block/{disk_name}"
+                )
+                if not result:
+                    logging.error(f"Disk {disk_path} not found on node {node}")
+                    raise ValueError(f"Disk {disk_path} not found")
+                
+                # Offline the disk to simulate full failure
+                runcommand.run(
+                    f"oc debug node/{node} -- chroot /host "
+                    f"echo offline > /sys/block/{disk_name}/state"
+                )
+                
+                # Monitor I/O impact
+                self.monitor_io(node, disk_path)
+                
+                # Wait for node to reflect failure (e.g., pod evictions, I/O errors)
+                nodeaction.wait_for_not_ready_status(node, timeout, self.kubecli, affected_node)
+                
+                logging.info(f"Full disk failure injected on {disk_path} for node {node}")
+                self.add_affected_node(affected_node)
+                
+            except Exception as e:
+                logging.error(f"Failed to simulate disk failure on {node}: {str(e)}")
+                raise e
+
+    def rollback_disk_failure(self, disk_path="/dev/sdb"):
+        """
+        Rollback the disk failure by restoring the disk to running state.
+        Args:
+            disk_path: Path to the disk (e.g., /dev/sdb).
+        """
+        for affected_node in self.affected_nodes_status.affected_nodes:
+            try:
+                logging.info(f"Rolling back disk failure on node {affected_node.node} for disk {disk_path}")
+                disk_name = disk_path.split('/')[-1]
+                runcommand.run(
+                    f"oc debug node/{affected_node.node} -- chroot /host "
```

Comment:
We try to keep krkn as kuberenetes based as possible, can you try using [this function](https://github.com/krkn-chaos/krkn-lib/blob/main/src/krkn_lib/k8s/krkn_kubernetes.py#L1071) in krkn-lib instead of oc debug to be sure this functionality can work on all cluster types